embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/SessionHandler.kt

```diff
@@ -233,126 +279,44 @@ internal class SessionHandler(
     }
 
     /**
-     * It builds an end active session message, it sanitizes it, it performs all types of memory cleaning,
-     * it updates cache and it sends it to our servers.
-     * It also stops periodic caching and automatic session stopper.
+     * 'Ends' the active session. Note that this logic is also used for caching the session
+     * periodically so the session won't always end. The behavior is controlled by the
+     * [SessionEndType] passed to this function.
      */
-    private fun runEndSessionFull(
-        endType: SessionLifeEventType,
-        originSession: Session?,
+    private fun runEndSessionImpl(
+        endType: SessionEndType,
+        activeSession: Session,
         sessionProperties: EmbraceSessionProperties,
         sdkStartupDuration: Long,
+        completedSpans: List<EmbraceSpanData>?,
+        lifeEventType: SessionLifeEventType,
```

Comment:
`SessionLifeEventType` is kind of confusing. Is there a way to refactor this out or somehow combine it with another concept? Have this and `SessionEndType` creates a number of permutations, many of which may not even be valid, which makes the code more complex than i needs to be (I hope)

Reply:
I have renamed `SessionEndType` to `SessionSnapshotType`. I can't think of a clean way to combine the two concepts as they are separate ideas: one is how the session was started; the other is how the session was written to disk.